src/connectors/current-refinements/connectCurrentRefinements.js

```diff
@@ -0,0 +1,307 @@
+  checkRendering(renderFn, usage);
+
+  return (widgetParams = {}) => {
+    if (widgetParams.includedAttributes && widgetParams.excludedAttributes) {
+      throw new Error(
+        '`includedAttributes` and `excludedAttributes` cannot be used together.'
+      );
+    }
+
+    const {
+      includedAttributes,
+      excludedAttributes = ['query'],
+      transformItems = items => items,
+    } = widgetParams;
+
+    return {
+      init({ helper, createURL, instantSearchInstance }) {
+        const refinements = transformItems(
+          getFilteredRefinements({
+            results: {},
+            state: helper.state,
+            helper,
+            includedAttributes,
+            excludedAttributes,
+          })
+        );
+
+        renderFn(
+          {
+            refine: refinement => clearRefinement(helper, refinement),
+            createURL: refinement =>
+              createURL(clearRefinementFromState(helper.state, refinement)),
+            refinements,
+            instantSearchInstance,
+            widgetParams,
+          },
+          true
+        );
+      },
+
+      render({ results, helper, state, createURL, instantSearchInstance }) {
+        const refinements = transformItems(
+          getFilteredRefinements({
+            results,
+            state,
+            helper,
+            includedAttributes,
+            excludedAttributes,
+          })
+        );
+
+        renderFn(
+          {
+            refine: refinement => clearRefinement(helper, refinement),
+            createURL: refinement =>
+              createURL(clearRefinementFromState(helper.state, refinement)),
+            refinements,
+            instantSearchInstance,
+            widgetParams,
+          },
+          false
+        );
+      },
+
+      dispose() {
+        unmountFn();
+      },
+    };
+  };
+}
+
+function getFilteredRefinements({
+  results,
+  state,
+  helper,
+  includedAttributes,
+  excludedAttributes,
+}) {
+  const clearsQuery =
+    (includedAttributes || []).indexOf('query') !== -1 ||
+    (excludedAttributes || []).indexOf('query') === -1;
+
+  const filterFunction = includedAttributes
+    ? refinement => includedAttributes.indexOf(refinement.attributeName) !== -1
+    : refinement => excludedAttributes.indexOf(refinement.attributeName) === -1;
+
+  const refinements = getRefinements(results, state, clearsQuery)
+    .filter(filterFunction)
+    .map(normalizeRefinementItem);
+
+  return groupByRefinements(refinements, helper);
+}
+
+function clearRefinementFromState(state, refinementItem) {
+  switch (refinementItem.type) {
+    case 'facet':
+      return state.removeFacetRefinement(
+        refinementItem.attribute,
+        refinementItem.value
+      );
+    case 'disjunctive':
+      return state.removeDisjunctiveFacetRefinement(
+        refinementItem.attribute,
+        refinementItem.value
+      );
+    case 'hierarchical':
+      return state.clearRefinements(refinementItem.attribute);
```

Comment:
Why are we using `clearRefinements` rather than `removeHierarchicalFacetRefinement`? The latter avoid to drop other refinements in case there are some on the same attribute.